stack.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"time"
+
+	"errors"
+
+	"github.com/distributeddesigns/currency"
+)
+
+type buyItem struct {
+	amount         currency.Currency
+	numStocks      uint
+	price          currency.Currency
+	stock          string
+	quoteTimeStamp time.Time
+}
+
+type buyStack []buyItem
+
+func (s buyStack) isEmpty() bool {
+	return len(s) == 0
+}
+
+func (bItem buyItem) isExpired() bool {
+	return bItem.quoteTimeStamp.Before(time.Now().Add(time.Second * -10))
+}
+
+func (s *buyStack) push(element buyItem) {
+	(*s) = append(*s, element)
+}
+
+func (s *buyStack) pop() (buyItem, error) {
+	var val buyItem
+	if (*s).isEmpty() == true {
+		return val, errors.New("Empty Stack")
+	}
+	val = (*s)[len(*s)-1]
+	*s = (*s)[:len(*s)-1]
+	return val, nil
+}
+
+func (s buyStack) peek() (buyItem, error) {
```

Comment:
Why does this function copy `s` but the others pass by reference? Seems like we'd always want pass by reference to avoid unnecessary memory allocation